src/components/utils/build-select-box-lists.js

```diff
@@ -0,0 +1,109 @@
+import { localeCompare, filterOsByArchitecture } from '_/helpers'
+
+/**
+ * Return a normalized and sorted list of data centers for use in a __SelectBox__ from
+ * the list of provided data centers.
+ */
+function createDataCenterList (dataCenters) {
+  const dataCenterList =
+    dataCenters
+      .toList()
+      .filter(datacenter => true) // TODO: canUserUseDataCenter permission check?
+      .map(dataCenter => ({
+        id: dataCenter.id,
+        value: dataCenter.name,
+      }))
+      .sort((a, b) => localeCompare(a.value, b.value))
+      .toJS()
+
+  return dataCenterList
+}
+
+/**
+ * Return a normalized and sorted list of clusters ready for use in a __SelectBox__ from
+ * the Map of provided clusters.
+ */
+function createClusterList (clusters, dataCenterId = false) {
```

Comment:
To be more consistent in types of argument, I think here would be better to use `dataCenterId = ''` or `dataCenterId = null`

Reply:
It's really 6 of one, half dozen of another.  The param will be evaluated as 'truthy' is one way or another.  Explicitly defaulting to null at least set a consistent interpretation.